src/main/java/me/oyurimatheus/nossomercadolivre/products/Product.java

```diff
@@ -0,0 +1,148 @@
+package me.oyurimatheus.nossomercadolivre.products;
+
+import me.oyurimatheus.nossomercadolivre.categories.Category;
+import me.oyurimatheus.nossomercadolivre.users.User;
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.validator.constraints.Length;
+
+import javax.persistence.*;
+import javax.validation.constraints.*;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.UUID;
+
+import static java.time.LocalDateTime.now;
+import static java.util.Objects.requireNonNull;
+import static org.springframework.util.Assert.hasText;
+import static org.springframework.util.Assert.notEmpty;
+
+@Table(name = "products")
+@Entity
+class Product {
+
+    @Id
+    @Column(name = "product_id")
+    private UUID id;
+
+    @NotBlank
+    @Column(name = "product_name")
+    private String name;
+
+    @DecimalMin(value = "0.01")
+    @Column(name = "product_price")
+    private BigDecimal price;
+
+    @Min(value = 0)
+    @Column(name = "product_stock_quantity")
+    private Integer stockQuantity;
+
+    @ElementCollection
+    @CollectionTable(name = "product_photos",
+            joinColumns = @JoinColumn(name = "product_id"))
+    @Size(min = 1)
+    private List<Photo> photos;
+
+    @ElementCollection
+    @CollectionTable(name = "product_characteristcs",
+            joinColumns = @JoinColumn(name = "product_id"))
+    @Size(min = 3)
+    private List<Characteristic> characteristics;
```

Comment:
Lembra, tem duas características iguais? Se não tem, pq você usou List?

Reply:
Força do hábito haha!